cmd2/cmd2.py

```diff
@@ -888,13 +902,30 @@ def _unregister_subcommands(self, cmdset: Union[CommandSet, 'Cmd']) -> None:
                     action.remove_parser(subcommand_name)
                     break
 
+    @property
+    def settables(self) -> Mapping[str, Settable]:
+        all_settables = dict(self._settables)
+        for cmd_set in self._installed_command_sets:
+            cmdset_settables = cmd_set.settables
+            for settable_name, settable in cmdset_settables.items():
+                if self.always_prefix_settables:
+                    all_settables[f'{cmd_set.settable_prefix}.{settable_name}'] = settable
+                else:
+                    all_settables[settable_name] = settable
+        return all_settables
+
     def add_settable(self, settable: Settable) -> None:
         """
         Convenience method to add a settable parameter to ``self.settables``
 
         :param settable: Settable object being added
         """
-        self.settables[settable.name] = settable
+        if not self.always_prefix_settables:
+            if settable.name in self.settables.keys() and settable.name not in self._settables.keys():
```

Comment:
Doesn't `self.settables` also include `self._settables`? Can the name be in one and not the other?

Reply:
Yes, because self.settables is now a property that aggregates settables in cmd2 and in each CommandSet.

Reply:
The settables property returns all settables in Cmd as well as all CommandSets. `self._settables` is only what's Cmd.

Reply:
Sorry, I get it now.